src/Compiler/Service/TransparentCompiler.fs

```diff
@@ -816,6 +818,14 @@ type internal TransparentCompiler
 
                 let bootstrapId = Interlocked.Increment &BootstrapInfoIdCounter
 
+                importsInvalidatedByTypeProvider.Publish.Add(fun () ->
+                    // caches.BootstrapInfoStatic.Clear()
+                    // caches.BootstrapInfo.Clear()
+                    caches.Clear(Set.singleton projectSnapshot.Identifier)
```

Comment:
If the project exposes the generated types in its public API (which we don't know, so probably have to assume it can..?), do we need to invalidate all projects that depend on it too?

Reply:
That is also the advantage of driving the invalidation just by the keys - you don't have to think about this. (At least here 😁)